### PR TITLE
kitty: 0.17.4 -> 0.18.0

### PR DESCRIPTION
**Pull request by Luflosi**
I submitted a patch to kitty, so that the library paths can now be passed as arguments to the build system, eliminating the need for patching. This should reduce the required maintenance effort in the future.
I removed an outdated comment.
I added a link to the changelog. Where is this link exposed to the user?
I added `dontConfigure = true;` to skip the configure phase since kitty has no configure script.
I now execute the test suite of kitty. Is the `doCheck = true;` attribute needed? It seems to work fine without but I added it anyways.
I also removed the dependency `which`.
Please check that the attributes are ordered in the preferred way.

###### Motivation for this change
https://github.com/kovidgoyal/kitty/releases/tag/v0.18.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tex** [comment]
Looks good to me. Can you squash commits into just one commit?

File: pkgs/applications/misc/kitty/default.nix [hunk 1]
```diff
@@ -63,25 +63,32 @@ buildPythonApplication rec {
...
   '' else ''
-    ${python.interpreter} setup.py linux-package --update-check-interval=0
+    ${python.interpreter} setup.py linux-package \
+    --update-check-interval=0 \
+    --egl-library='${stdenv.lib.getLib libGL}/lib/libEGL.so.1' \
+    --startup-notification-library='${libstartup_notification}/lib/libstartup-notification-1.so' \
+    --canberra-library='${libcanberra}/lib/libcanberra.so'
+  '';
+
+  doCheck = true;
```
Comment (teto) [inline pkgs/applications/misc/kitty/default.nix:86]: unneeded, it is true by default.

File: pkgs/applications/misc/kitty/default.nix [hunk 2]
```diff
@@ -125,6 +130,7 @@ buildPythonApplication rec {
     homepage = "https://github.com/kovidgoyal/kitty";
     description = "A modern, hackable, featureful, OpenGL based terminal emulator";
     license = licenses.gpl3;
+    changelog = "https://sw.kovidgoyal.net/kitty/changelog.html";
```
Comment (teto) [inline pkgs/applications/misc/kitty/default.nix:133]: didn't know about this meta entry: cool !!

**Review by teto — APPROVED** [review]
passed nixpkgs-review and binary worked fine. Maintainers can squash. Thanks for making kitty easier to maintain.

**Review by tex — APPROVED** [review]

File: pkgs/applications/misc/kitty/default.nix (hunk 1 above)
Comment (Luflosi) [reply pkgs/applications/misc/kitty/default.nix]: I removed it.

**Comment by Luflosi** [comment]
I put all the changes into one commit and rebased onto the latest master.

**Comment by Luflosi** [comment]
Thanks for merging ❤️.

**Pull request merged**